<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ARBwaveformEdit</class>
 <widget class="QDialog" name="ARBwaveformEdit">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>617</width>
    <height>494</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Arbirtary Waveform Editor</string>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>455</x>
     <y>450</y>
     <width>151</width>
     <height>32</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Press OK to accept the waveform changes and download to the connected MIPS system.</string>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="txtData">
   <property name="geometry">
    <rect>
     <x>15</x>
     <y>85</y>
     <width>181</width>
     <height>396</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Waveform data table. 32 total entries, one per line. Valid range is -100 to 100.</string>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>5</y>
     <width>171</width>
     <height>66</height>
    </rect>
   </property>
   <property name="text">
    <string>Arbitrary waveform data. Enter/edit values from -100 to 100 in the table below.</string>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>10</y>
     <width>351</width>
     <height>121</height>
    </rect>
   </property>
   <property name="title">
    <string>Waveform generation options</string>
   </property>
   <widget class="QLineEdit" name="leUp">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>35</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>4</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>35</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Up</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>35</y>
      <width>156</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Down, square waveform</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pbGenUpDown">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>30</y>
      <width>91</width>
      <height>32</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Generates a waveform and plots based on the Up and Down parameters defined. Up plus Down must be &gt; 2 and less that 32.</string>
    </property>
    <property name="text">
     <string>Generate</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>65</y>
      <width>171</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Sine, with starting phase =</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lePhase">
    <property name="geometry">
     <rect>
      <x>185</x>
      <y>65</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pbGenSine">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>60</y>
      <width>91</width>
      <height>32</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Generates a sine wave with user defined initial phase in degrees.</string>
    </property>
    <property name="text">
     <string>Generate</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="leDown">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>35</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>4</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>171</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Cycles per waveform</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="leCycles">
    <property name="geometry">
     <rect>
      <x>185</x>
      <y>90</y>
      <width>60</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>1</string>
    </property>
   </widget>
  </widget>
  <widget class="QPushButton" name="pbPlot">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>450</y>
     <width>121</width>
     <height>32</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Plots the waveform definded by the data in the wwaveform table.</string>
   </property>
   <property name="text">
    <string>Plot waveform</string>
   </property>
  </widget>
  <widget class="QCustomPlot" name="plot" native="true">
   <property name="geometry">
    <rect>
     <x>215</x>
     <y>145</y>
     <width>386</width>
     <height>291</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="pbInvert">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>450</y>
     <width>131</width>
     <height>32</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Changes the sign of every entry in the data table and then plots the waveform.</string>
   </property>
   <property name="text">
    <string>Invert waveform</string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ARBwaveformEdit</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ARBwaveformEdit</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
